src/service.rs

```diff
@@ -1535,6 +1548,32 @@ impl Service {
         }
     }
 
+    /// This is a helper function that determines if we need more votes for a specific IP
+    /// class.
+    ///
+    /// If we are in dual-stack made and don't have enough votes for either ipv4 or ipv6 and the
+    /// requesting node/vote is what we need, then this will return true
+    fn require_more_ip_votes(&mut self, is_ipv6: bool) -> bool {
+        if !matches!(self.ip_mode, IpMode::DualStack) {
+            return false;
+        }
+
+        if let Some(ip_votes) = self.ip_votes.as_mut() {
```

Comment:
nit:
```suggestion
       let Some(ip_votes) = self.ip_votes.as_mut() else {
           return false;
       }
```
instead of returning false after checking the majority and `is_ipv6` conditions, reducing the scope of the if condition of `ip_votes` being `None` and returning early.

Reply:
I think this change is the same. 

If self.ip_votes is None, we immediately return false, we don't check `is_ipv6` or the majority. 

Adding this PR because its probably easier to see this